cmake/kokkos_enable_options.cmake

```diff
@@ -53,6 +53,50 @@ MESSAGE(STATUS "Setting ${ENABLE_NAME}=ON from KOKKOS_DEVICES")
 SET(${ENABLE_NAME} ON CACHE BOOL "Enable device ${DEV}" FORCE)
 ENDFOREACH()
 
+IF (KOKKOS_ARCH MATCHES ",")
+  MESSAGE(WARNING "-- Detected a comma in: KOKKOS_ARCH=`${KOKKOS_ARCH}`")
+  MESSAGE("-- Although we prefer KOKKOS_ARCH to be semicolon-delimited, we do allow")
+  MESSAGE("-- comma-delimited values for compatibility with scripts (see github.com/trilinos/Trilinos/issues/2330)")
+  STRING(REPLACE "," ";" KOKKOS_ARCH "${KOKKOS_ARCH}")
+  MESSAGE("-- Commas were changed to semicolons, now KOKKOS_ARCH=`${KOKKOS_ARCH}`")
+ENDIF()
```

Comment:
Just wanted to mention here another reason one would prefer to use commas:
When passing say `-DKOKKOS_ARCH=Volta70,Power9` to cmake I don't have to use double quotes which I would then need to escape if I define a variable with all the options I want to pass to configure Kokkos.